<?xml version="1.0"?>
<robot name="hand">

  <material name="grey">
    <color rgba="0.5 0.5 0.5 1.0"/>
  </material>

  <link name="base_link">
    <inertial>
      <mass value="0.6"/>
      <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.04 0.2 0.08"/>
      </geometry>
      <material name="grey"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.04 0.2 0.08"/>
      </geometry>
      <material name="grey"/>
    </collision>
    <contact>
      <lateral_friction value="0.8"/>
      <spinning_friction value="0.1"/>
    </contact>
  </link>

  <!-- joint between base and left finger -->
  <joint name="finger_l" type="prismatic">
    <parent link="base_link"/>
    <child link="finger_l_link"/>
    <origin rpy="0 0 0" xyz="0 -0.01 0.065"/>
    <axis xyz="0 -1 0"/>
    <limit lower="0" upper="0.04" effort="20" velocity="0.02"/>
    <dynamics damping="0.5"/>
  </joint>

  <link name="finger_l_link">
    <inertial>
      <mass value="0.06"/>
      <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.02 0.02 0.05"/>
      </geometry>
      <material name="grey"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.02 0.02 0.05"/>
      </geometry>
      <material name="grey"/>
    </collision>
    <contact>
      <lateral_friction value="0.8"/>
      <spinning_friction value="0.1"/>
    </contact>
  </link>

  <!-- joint between base and right finger -->
  <joint name="finger_r" type="prismatic">
    <parent link="base_link"/>
    <child link="finger_r_link"/>
    <origin rpy="0 0 0" xyz="0.0 0.01 0.065"/>
    <axis xyz="0 1 0"/>
    <limit lower="0" upper="0.04" effort="20" velocity="0.02"/>
    <dynamics damping="0.5"/>
  </joint>

  <link name="finger_r_link">
    <inertial>
      <mass value="0.06"/>
      <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="0"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.02 0.02 0.05"/>
      </geometry>
      <material name="grey"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.02 0.02 0.05"/>
      </geometry>
      <material name="grey"/>
    </collision>
    <contact>
      <lateral_friction value="0.8"/>
      <spinning_friction value="0.1"/>
    </contact>
  </link>

</robot>
